CONTRIBUTING.md

```diff
@@ -8,18 +8,22 @@ Thank your for contributing to this project! We welcome collaborators and expect
 4. Run Rustfmt, Clippy, and all tests to ensure CI rules are satisfied. Correct versions and feature flags can be found in the [`.travis.yml`](https://github.com/poanetwork/hbbft/blob/master/.travis.yml) file.
 5. Commit your changes (`git commit -am 'Add some feature'`).
 6. Push to your branch (`git push origin my-new-feature`).
-7. Create a new Pull Request.
+7. Create a new PR (Pull Request).
 
 ### General
 
+* We strive to follow the [Rust API Guidelines](https://rust-lang-nursery.github.io/api-guidelines/about.html) to maintain consistency and compatibility in our code.     
 * Commits should be one logical change that still allows all tests to pass.  We prefer smaller commits if there could be two levels of logic grouping.  The goal is to provide future contributors (including your future self) the reasoning behind your changes and allow them to cherry-pick, patch or port those changes in isolation to other branches or forks.
 * If during your PR you reveal a pre-existing bug:
   1. Try to isolate the bug and fix it on an independent branch and PR it first.
-  2. Try to fix the bug in a separate commit from other changes:
-     1. Commit the code in the broken state that revealed the bug originally
+  2. Try to fix the bug in a separate commit from other changes.
+     1. Commit the code in the broken state that revealed the bug originally.
```

Comment:
I think the most common case is really that someone finds a bug and doesn't know how to fix it. Which is fine, as documented below. But maybe we should add "you reveal a pre-existing bug and know how to fix it" to line 17?

And then maybe I'd just make two points in total:

```markdown
1. If you can isolate the bug, fix it in a separate PR.
2. If it depends on your other commits, add the fix in a separate commit to the same PR.

In either case, try to write a regression test that would fail because of the bug, but passes with your fix.
```